pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantExporter.java

```diff
@@ -81,75 +134,100 @@ public static List<String> getAllColumnNames(FilterExportConfig filterConfig, in
                 activityName + COLUMN_REPEAT_DELIMITER + activityRepeatNum : activityName;
         if (filterConfig.getColumn().getObject() != null) {
             activityName = activityName + DBConstants.ALIAS_DELIMITER + filterConfig.getColumn().getObject();
         }
+        String questionStableId = filterConfig.getColumn().getName();
         String columnExportName = questionRepeatNum > 1 ?
                 questionStableId + COLUMN_REPEAT_DELIMITER + questionRepeatNum : questionStableId;
-        if (hasSeparateColumnForOption(option, filterConfig)) {
-            columnExportName = columnExportName + DBConstants.ALIAS_DELIMITER + option.get(ESObjectConstants.OPTION_STABLE_ID);
+
+        if (childConfig != null) {
+            columnExportName = columnExportName + DBConstants.ALIAS_DELIMITER + childConfig.getColumn().getName();
+        }
+        if (hasSeparateColumnForOption(optionStableId, filterConfig)) {
+            columnExportName = columnExportName + DBConstants.ALIAS_DELIMITER + optionStableId;
         }
-        if (columnExportName.endsWith("REGISTRATION_STATE_PROVINCE")) {
-            columnExportName = "REGISTRATION_STATE_PROVINCE";
+
+        if (detailName != null) {
+            columnExportName = columnExportName + OPTION_DETAIL_DELIMITER + detailName;
+        }
+
+        // if this is a collated column, use the suffix (e.g. use "REGISTRATION_STATE_PROVINCE" instead of "CA_REGISTRATION_STATE_PROVINCE")
+        for (String suffix : ValueProviderFactory.COLLATED_SUFFIXES) {
+            if (columnExportName.endsWith(suffix)) {
+                columnExportName = suffix;
+            }
         }
+
         String exportName = activityExportName + DBConstants.ALIAS_DELIMITER + columnExportName;
-        return exportName;
+        return exportName.toUpperCase();
     }
 
-    private static boolean hasSeparateColumnForOption(Map<String, Object> option, FilterExportConfig filterConfig) {
-        return option != null && filterConfig.isSplitOptionsIntoColumns();
+    /** whether a separate column will be rendered for the option */
+    private static boolean hasSeparateColumnForOption(String optionStableId, FilterExportConfig filterConfig) {
+        return optionStableId != null && filterConfig.isSplitOptionsIntoColumns();
     }
 
+    /** gets the header row -- this iterates through all the configs and calls "getAllColumnNames" for each */
     protected List<String> getHeaderRow() {
         List<String> headers = new ArrayList<>();
         for (ModuleExportConfig moduleConfig : moduleConfigs) {
             for (int formRepeatNum = 1; formRepeatNum <= moduleConfig.getNumMaxRepeats(); formRepeatNum++) {
                 for (FilterExportConfig filterConfig : moduleConfig.getQuestions()) {
-                    headers.addAll(getAllColumnNames(filterConfig, formRepeatNum, 1));
+                    for (int questionRepeatNum = 1; questionRepeatNum <= filterConfig.getMaxRepeats(); questionRepeatNum++) {
```

Comment:
Just wondering, having such many nested for loops how badly will affect the performance?

Reply:
yeah, `for` loops are AFAIK the fastest way to iterate, and since we have to output everything, there's no escaping the loops here.  Fortunately, we only execute this method once for the entire export operation, so it's not a bottleneck